File: 알고리즘/최이삭/Origami.js
function solution(n) {
  let answer = [0];
  for(let i = 1; i < n; i++) {
      let answerLength = answer.length;
      let hello = answer.map((el, idx) => {
          if(Math.floor(answerLength / 2) === idx) return 1; // 중앙을 1로
          else return el;
      }) 
      
      answer = [...answer, 0, ...hello] // 답
  }
  return answer;
}

File: 알고리즘/김기강/20.06.03 origami/Origami.java
class Solution {
    public int[] solution(int n) {
        int len = (int)Math.pow(2, n)-1;    //double 값이라 (int)~
        int[] answer = new int[len];
        
        String ansS = "0";
        String updown;
        for(int i = 2; i <= n; i++) {
            updown = "1";
            StringBuilder space = new StringBuilder("0");
            for(int j = 0; j < ansS.length(); j++) {
                space.append(ansS.charAt(j) + updown);
                updown = updown.equals("1") ? "0" : "1";
            }
            ansS = space.toString();
        }
        
        for(int i = 0; i < ansS.length(); i++) {    //int로 바꿔서 넣어줌
            answer[i] = ansS.charAt(i) - 48;    //char 0 == int 48, char 1 == int 49
        }
        
        return answer;
    }
}